### PR TITLE
Better clickhouse-client multiline input

### PR DESCRIPTION
**Pull request by amosbird**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Enable multiline editing in clickhouse-client by default. This addresses https://github.com/ClickHouse/ClickHouse/issues/31121 .

Before https://github.com/ClickHouse/ClickHouse/pull/30143 , we didn't support native multline editing in clickhouse-client. Thus bracketed pasting is only enabled when `-n` is enabled and `-m` is disabled (see https://github.com/ClickHouse/ClickHouse/issues/7757) . Now we allow multiple sections of multiline inputs delimited by extender character: `\`.

Detailed description / Documentation draft:
.

### CODE REVIEW

**Comment by amosbird** [comment]
“Timeout dropping database test_3gkcv6 after test”? but why

File: tests/queries/0_stateless/01526_client_start_and_exit.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ ${CLICKHOUSE_CLIENT} -q "SELECT 'CREATE TABLE test_' || hex(randomPrintableASCII
 function stress()
 {
     while true; do
-        "${CURDIR}"/01526_client_start_and_exit.expect-not-a-test-case | grep -v -P 'ClickHouse client|Connecting|Connected|:\) Bye\.|new year|^\s*$|spawn bash|^0\s*$'
+        "${CURDIR}"/01526_client_start_and_exit.expect-not-a-test-case | grep -v -P 'ClickHouse client|Connecting|Connected|:\) Bye\.|new year|^\s*$|spawn bash|\?2004l|^0\s*$'
```
Comment (novikd) [inline tests/queries/0_stateless/01526_client_start_and_exit.sh:14]: I didn't understand why this change checks correctness. Could you please explain it?

Comment (amosbird) [reply tests/queries/0_stateless/01526_client_start_and_exit.sh:14]: It's most likely because parallel running `expect` emulated terminal doesn't work well with bracketed paste enabling sequence, which is `\e033?2004l`  (FYI https://cirw.in/blog/bracketed-paste)

Comment (alexey-milovidov) [reply tests/queries/0_stateless/01526_client_start_and_exit.sh:14]: Then write a comment inside the test, otherwise it's very obscure.

**Review by alexey-milovidov — APPROVED** [review]
